CIPs/0022.md

```diff
@@ -0,0 +1,40 @@
+---
+cip: 0022
+title: Standard for Injacted web3 Provider
+author: @daoauth @NAKsir-melody
+discussions-to: https://github.com/celo-org/celo-proposals/issues/74
+status: Draft
+type:
+category:
+created: 2020-10-29
+license: Apache 2.0
+---
+
+## Simple Summary
+Standard to find out the type of web3 provider to support dapps running in web browsers
+
+## Abstract
+- In the case of web DApps, you can only use [DSRV Celo Extension Wallet](https://chrome.google.com/webstore/detail/celo-desktop-wallet/kkilomkmpmkbdnfelcpgckmpcaemjcdh) or MetaMask Extension wallet, but a standard is required to distinguish them.
+- Through these standards, each provider can be identified and transactions can be sent to the Celo blockchain through appropriate operations.
+
+## Motivation
+To increase usability and accessibility by allowing celo DApps to be used not only in mobile environment but also in desktop environment.
+
+## Specification
+- Ready `ethereum.isMetaMask` property for MetaMask. Through this property, it can be seen that it is a provider injected in [MetaMask](https://docs.metamask.io/guide/ethereum-provider.html#properties).
+- So, it is suggested that providers injected in Celo have a property with the syntax of `celo[providerName]`.
+  * eg. Some injected provider has `celo.isDesktop` property. It's injected by DSRV Celo Extension wallet
+  * eg. For DApp browser has `celo.isMobile` too
+
+## Test Cases
+
+
+## Implementation
+- [Celo web signer](https://github.com/dexfair/celo-web-signer)
+- [Celo remix plugin](https://github.com/dexfair/celo-remix-plugin)
+- [DSRV Celo Extension Wallet](https://github.com/dsrvlabs/celo-extension-wallet)
+
+## Security Considerations
```

Comment:
Any security considerations we should be thinking about here with the addition of new RPC endpoints?

Reply:
Since it is a proposal to use MetaMask and other web extension wallets(eg. DSRV's extension wallet, DApp browser), so The commands of transaction or sign message are done through web3 like general DApps. so there is no security for this proposal.